pom.xml

```diff
@@ -411,6 +411,12 @@
       <version>${forbiddenapis.version}</version>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.googlejavaformat</groupId>
+      <artifactId>google-java-format</artifactId>
+      <version>1.22.0</version>
+      <scope>test</scope>
+    </dependency>
```

Comment:
Added the google-java-format as a dependency

Reply:
Error from CI:
> java.lang.IllegalAccessError: class com.google.googlejavaformat.java.JavaInput (in unnamed module @0x459e9125) cannot access class com.sun.tools.javac.parser.Tokens$TokenKind (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.parser to unnamed module @0x459e9125

Reply:
looks like we need something like:
https://github.com/Spotnana-Tech/google-style-precommit-hook/pull/2/files

there is huge issue on this plugin https://github.com/google/google-java-format/issues/787

looks like we need to place such parameters at file
https://github.com/Cosium/git-code-format-maven-plugin?tab=readme-ov-file#jdk-16-peculiarities

if we plan to avoid such hacks, we can move formatting to plugin execution and run plugin only in CI in special profile that has all such extra parameters for jvm.

Reply:
still active issue: https://github.com/google/google-java-format/issues/1079

Reply:
we need to add arguments at https://github.com/checkstyle/checkstyle/blob/947002a3d66557d7e62a5aee81c7de449ce5f168/azure-pipelines.yml#L99-L101

Reply:
we can experiment on small PR like https://github.com/checkstyle/checkstyle/pull/15210/files

Reply:
@Zopsss , please use fix from https://github.com/checkstyle/checkstyle/pull/15227 to bypass this problem